src/Microsoft.AspNet.Mvc.ModelBinding/Internal/DictionaryHelper.cs

```diff
@@ -28,6 +29,20 @@ public static class DictionaryHelper
                     continue;
                 }
 
+                if (key.StartsWith("[", StringComparison.OrdinalIgnoreCase))
+                {
+                    var splitKey = key.Split('.');
+                    if (splitKey.Count() > 1)
+                    {
+                        key = splitKey[1];
+                        if (string.Equals(prefix, key, StringComparison.Ordinal))
```

Comment:
Synced up Offline. Adding `key.StartsWith` will return entries like `ProductName` for a prefix `Product`. This is not valid. So I will go with `string.Equals` here.